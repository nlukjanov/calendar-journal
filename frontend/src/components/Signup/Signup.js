import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import axios from 'axios';

const Signup = () => {
  const history = useHistory();
  const [formData, setFormData] = useState();
  const [signupErrors, setSignupErrors] = useState();

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await axios.post('http://localhost:4000/api/register', formData);
      history.push('/signin');
      // add notification with res
    } catch (error) {
      setSignupErrors({ errors: error.response.data.errors });
    }
  };

  // logging after state update
  useEffect(() => console.log(signupErrors?.errors), [signupErrors]);

  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='username'>
          <Form.Label>Username</Form.Label>
          <Form.Control
            name='username'
            type='username'
            placeholder='Enter username'
            onChange={handleChange}
          />
          {signupErrors && <small>{signupErrors?.errors.username}</small>}
        </Form.Group>
        <Form.Group controlId='email'>
          <Form.Label>Email</Form.Label>
          <Form.Control
            name='email'
            type='email'
            placeholder='Enter email'
            onChange={handleChange}
          />
          {signupErrors && <small>{signupErrors?.errors.email}</small>}
          <Form.Text className='text-muted'>
            We never share your email with anyone else.
          </Form.Text>
        </Form.Group>
        <Form.Group controlId='password'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='password'
            placeholder='Password'
            onChange={handleChange}
            autoComplete='on'
          />
          {signupErrors && <small>{signupErrors?.errors.password}</small>}
        </Form.Group>
        <Form.Group controlId='passwordConfirmation'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='passwordConfirmation'
            placeholder='Password Confirmation'
            onChange={handleChange}
            autoComplete='on'
          />
          {signupErrors && (
            <div>
              <small>{signupErrors?.errors.passwordConfirmation}</small>
            </div>
          )}
        </Form.Group>
        <Button variant='primary' type='submit'>
          Sign Up
        </Button>
      </Form>
    </Card>
  );
};

export default Signup;
