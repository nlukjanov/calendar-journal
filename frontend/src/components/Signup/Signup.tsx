import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import axios from 'axios';

const Signup = () => {
  const history = useHistory();
  const [formData, setFormData] = useState({});
  const [signupErrors, setSignupErrors] = useState<Error | null>(null);

  type Change = {
    target: Field;
  };

  type Field = {
    name: string;
    value: string;
  };

  type Error = {
    errors: Errors;
  };

  type Errors = {
    response: Response;
    prevState: null;
    username: string;
    email: string;
    password: string;
    passwordConfirmation: string;
  };

  type Response = {
    data: Data;
  };

  type Data = {
    message: string;
  };

  const handleChange = ({ target: { name, value } }: Change) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  const handleSubmit = async (e: any) => {
    e.preventDefault();
    try {
      await axios.post('http://localhost:4000/api/register', formData);
      history.push('/signin');
      // add notification with res
    } catch (error) {
      setSignupErrors({ errors: error.response.data.errors });
    }
  };

  // logging after state update
  useEffect(() => console.log(signupErrors?.errors), [signupErrors]);

  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='username'>
          <Form.Label>Username</Form.Label>
          <Form.Control
            name='username'
            type='username'
            placeholder='Enter username'
            onChange={handleChange}
          />
          {signupErrors && <small>{signupErrors?.errors.username}</small>}
        </Form.Group>
        <Form.Group controlId='email'>
          <Form.Label>Email</Form.Label>
          <Form.Control
            name='email'
            type='email'
            placeholder='Enter email'
            onChange={handleChange}
          />
          {signupErrors && <small>{signupErrors?.errors.email}</small>}
          <Form.Text className='text-muted'>
            We never share your email with anyone else.
          </Form.Text>
        </Form.Group>
        <Form.Group controlId='password'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='password'
            placeholder='Password'
            onChange={handleChange}
            autoComplete='on'
          />
          {signupErrors && <small>{signupErrors?.errors.password}</small>}
        </Form.Group>
        <Form.Group controlId='passwordConfirmation'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='passwordConfirmation'
            placeholder='Password Confirmation'
            onChange={handleChange}
            autoComplete='on'
          />
          {signupErrors && (
            <div>
              <small>{signupErrors?.errors.passwordConfirmation}</small>
            </div>
          )}
        </Form.Group>
        <Button variant='primary' type='submit'>
          Sign Up
        </Button>
      </Form>
    </Card>
  );
};

export default Signup;
