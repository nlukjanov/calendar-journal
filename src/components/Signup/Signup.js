import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';

const Signup = () => {
  const history = useHistory();
  const [formData, setFormData] = useState();
  const [signupErrors, setSignupErrors] = useState();

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    console.log(formData);
    try {
      await axios.post('http://localhost:4000/api/register', formData);
      history.push('/signin');
      // add notification with res

    } catch (error) {
      console.log(error);
      setSignupErrors(error);
    }
  };

  // logging after state update
  useEffect(() => console.log(formData), [formData]);

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor='username'>Username</label>
          <input
            id='username'
            type='text'
            name='username'
            placeholder='Username'
            onChange={handleChange}
          />
        </div>
        <div>
          <label htmlFor='email'>Email</label>
          <input
            id='email'
            type='text'
            name='email'
            placeholder='Email'
            onChange={handleChange}
          />
        </div>
        <div>
          <label htmlFor='password'>Password</label>
          <input
            id='password'
            type='password'
            name='password'
            placeholder='Password'
            autoComplete='on'
            onChange={handleChange}
          />
        </div>
        <div>
          <label htmlFor='passwordConfirmation'>Password Confirmation</label>
          <input
            id='passwordConfirmation'
            type='password'
            name='passwordConfirmation'
            placeholder='Password Confirmation'
            autoComplete='on'
            onChange={handleChange}
          />
        </div>
        <div>
          <button type='submit'>Sign Up</button>
        </div>
      </form>
    </div>
  );
};

export default Signup;
