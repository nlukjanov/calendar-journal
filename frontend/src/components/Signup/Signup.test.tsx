import React from 'react';

import { render } from '@testing-library/react';
import Signup from './Signup';

test('renders new journal entry', () => {
  const { container } = render(<Signup />);
  expect(container).toMatchSnapshot();
});
