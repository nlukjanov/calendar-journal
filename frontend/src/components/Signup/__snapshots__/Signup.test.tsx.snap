// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new journal entry 1`] = `
<div>
  <div
    class="mx-auto mt-5 border-light card"
    style="max-width: 25rem;"
  >
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="username"
        >
          Username
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="Enter username"
          type="username"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="email"
        >
          Email
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          placeholder="Enter email"
          type="email"
        />
        <small
          class="text-muted form-text"
        >
          We never share your email with anyone else.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="on"
          class="form-control"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="passwordConfirmation"
        >
          Password
        </label>
        <input
          autocomplete="on"
          class="form-control"
          id="passwordConfirmation"
          name="passwordConfirmation"
          placeholder="Password Confirmation"
          type="password"
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
